src/core/server/saved_objects/routes/import.test.ts

```diff
@@ -0,0 +1,320 @@
+
+describe('POST /api/saved_objects/_import', () => {
+  let server: setupServerReturn['server'];
+  let httpSetup: setupServerReturn['httpSetup'];
+  let handlerContext: setupServerReturn['handlerContext'];
+  let savedObjectsClient: ReturnType<typeof savedObjectsClientMock.create>;
+
+  const emptyResponse = {
+    saved_objects: [],
+    total: 0,
+    per_page: 0,
+    page: 0,
+  };
+
+  beforeEach(async () => {
+    ({ server, httpSetup, handlerContext } = await setupServer());
+    savedObjectsClient = handlerContext.savedObjects.client;
+
+    savedObjectsClient.find.mockResolvedValue(emptyResponse);
+
+    const router = httpSetup.createRouter('/api/saved_objects/');
+    registerImportRoute(router, config, allowedTypes);
+
+    await server.start();
+  });
+
+  afterEach(async () => {
+    await server.stop();
+  });
+
+  it('formats successful response', async () => {
+    const result = await supertest(httpSetup.server.listener)
+      .post('/api/saved_objects/_import')
+      .set('content-Type', 'multipart/form-data; boundary=BOUNDARY')
+      .send(
+        [
+          '--BOUNDARY',
+          'Content-Disposition: form-data; name="file"; filename="export.ndjson"',
+          'Content-Type: application/ndjson',
+          '',
+          '',
+          '--BOUNDARY--',
+        ].join('\r\n')
+      )
+      .expect(200);
+
+    expect(result.body).toEqual({
+      success: true,
+      successCount: 0,
+    });
+    expect(savedObjectsClient.bulkCreate).toHaveBeenCalledTimes(0);
+  });
+
+  it('defaults migrationVersion to empty object', async () => {
+    savedObjectsClient.bulkCreate.mockResolvedValueOnce({
+      saved_objects: [
+        {
+          type: 'index-pattern',
+          id: 'my-pattern',
+          attributes: {
+            title: 'my-pattern-*',
+          },
+          references: [],
+        },
+      ],
+    });
+
+    const result = await supertest(httpSetup.server.listener)
+      .post('/api/saved_objects/_import')
+      .set('content-Type', 'multipart/form-data; boundary=EXAMPLE')
+      .send(
+        [
+          '--EXAMPLE',
+          'Content-Disposition: form-data; name="file"; filename="export.ndjson"',
+          'Content-Type: application/ndjson',
+          '',
+          '{"type":"index-pattern","id":"my-pattern","attributes":{"title":"my-pattern-*"}}',
+          '--EXAMPLE--',
+        ].join('\r\n')
+      )
+      .expect(200);
+
+    expect(result.body).toEqual({
+      success: true,
+      successCount: 1,
+    });
+    expect(savedObjectsClient.bulkCreate).toHaveBeenCalledTimes(1);
+    const firstBulkCreateCallArray = savedObjectsClient.bulkCreate.mock.calls[0][0];
+    expect(firstBulkCreateCallArray).toHaveLength(1);
+    expect(firstBulkCreateCallArray[0].migrationVersion).toEqual({});
+  });
+
+  it('imports an index pattern and dashboard, ignoring empty lines in the file', async () => {
+    // NOTE: changes to this scenario should be reflected in the docs
```

Comment:
Does the comment mean it's the public API?

Reply:
I kept the comments untouched when I moved/adapted the tests.

The API is available here: https://www.elastic.co/guide/en/kibana/master/saved-objects-api.html, so it seems 'public', but everything is flagged as experimental.